File: tradehelper-ui/src/app/shared/models/binancesession.ts
import { Exchange } from "../enums/exchange";
import { ISession } from "../interfaces/isession";

export class BinanceSession implements ISession {

    constructor(){
        this.id = "";
        this.name= "";
        this.exchange = Exchange.DEFAULT;
        this.key = "";
        this.secret = "";
    }

    id: string;
    name: string;
    exchange: Exchange;
    key: string;
    secret: string;
}

File: tradehelper-api/src/main/java/com/tradehelper/api/models/BinanceSession.java
package com.tradehelper.api.models;

import java.io.Serializable;

public class BinanceSession extends Session implements Serializable {
	private String key;
	private String secret;

	public BinanceSession() {
		super("", "", null);
		this.key = "";
		this.secret = "";
	}

	public BinanceSession(String id, String name, Exchange exchange, String key, String secret) {
		super(id, name, exchange);
		this.key = key;
		this.secret = secret;
	}

	public String getKey() {
		return key;
	}

	public void setKey(String key) {
		this.key = key;
	}

	public String getSecret() {
		return secret;
	}

	public void setSecret(String secret) {
		this.secret = secret;
	}

}
